opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -229,8 +229,14 @@ def get_tracer(
         schema_url: typing.Optional[str] = None,
     ) -> "Tracer":
         # pylint:disable=no-self-use,unused-argument
-        return _DefaultTracer()
+        return _NoOpTracer()
+
+@deprecated(version="1.9.0", reason="You should use _NoOpTracerProvider")  # type: ignore
```

Comment:
Also, I have my doubts on making private a class that is "publicly" exposed in an entry point :thinking:

Reply:
I would be ok with making the NoOpTracer/TracerProvider public. What do others think? @open-telemetry/python-approvers

Reply:
I think the `NoOpTracerX` should be public :+1:

Reply:
I am ok with this too.

Reply:
I think it is perfectly fine to have a private implementation as an entry point value but fine with making it public as well.